lib/ae_mdw_web/controllers/tx_controller.ex

```diff
@@ -106,22 +106,23 @@ defmodule AeMdwWeb.TxController do
   defp handle_spendtx_details(response_data, %{"account" => _account}) do
     response_data
     |> Enum.uniq_by(fn %{"hash" => tx_hash} -> tx_hash end)
-    |> Enum.map(fn %{"tx" => block_tx, "tx_index" => tx_index} = block ->
-      spend_tx_recipient = (block_tx["type"] == @type_spend_tx && block_tx["recipient_id"]) || nil
-      add_details? = String.starts_with?(spend_tx_recipient || "", "nm_")
-
-      if add_details? do
-        update_in(block, ["tx"], fn block_tx ->
-          Map.merge(block_tx, get_recipient(spend_tx_recipient, tx_index))
-        end)
-      else
-        block
-      end
-    end)
+    |> Enum.map(&maybe_add_spend_tx_details/1)
   end
 
   defp handle_spendtx_details(response_data, _params), do: response_data
 
+  defp maybe_add_spend_tx_details(%{"tx" => block_tx, "tx_index" => tx_index} = block) do
+    recipient_id = block_tx["recipient_id"] || ""
+
+    if block_tx["type"] == @type_spend_tx and String.starts_with?(recipient_id, "nm_") do
```

Comment:
do we usually do these checks with String methods? Or can't we use the ID system from the node in binary?

Reply:
We really need to test the formatted recipient_id here because of multiple sources.